(WIP) UDF Schema Prototype

chengxiong-ruan opened the pull request:
Prototyping the SQL grammar of `CREATE FUNCTION` statement.

Release note: None.

chengxiong-ruan committed:
sql: `CREATE FUNCTION` sql grammar prototype

Prototyping the SQL grammar of `CREATE FUNCTION` statement.

Release note: None.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/81688)
<!-- Reviewable:end -->

chengxiong-ruan committed:
sql: create function in legacy schema changer
making it able to store a udf descriptor through `CREATE FUNCTION` statement.
Release note: None.

chengxiong-ruan committed:
sql: toy function resolver

this commit added a function resolver, which currently ignore the cache.
also integrated it to a few places functions need to be resolved.
but I'm not familiar with execution stuff...but the name resolving should
work for now (without cache).

Release note: None.

chengxiong-ruan committed:
sql: toy udf execution

copy of Marcus's execution prototype in #82425

Release note: None.

chengxiong-ruan committed:
sql: connect udf schema toy and query execution toy

Release note: None.
